astropy/modeling/core.py

```diff
@@ -1585,11 +1585,17 @@ def _process_output_units(self, inputs, outputs):
 
         return outputs
 
-    def prepare_outputs(self, format_info, *outputs, **kwargs):
+    def prepare_outputs(self, format_info, *outputs, model_set_axis=None, **kwargs):
+        if model_set_axis is None:
+            # By default the model_set_axis for the input is assumed to be the
+            # same as that for the parameters the model was defined with
+            # TODO: Ensure that negative model_set_axis arguments are respected
```

Comment:
Is this something that you plan to address in this PR or for the future?

Reply:
There are many `TODO` comments which don't seem to match the code. This one I think was working before and I tried to keep it working.  Will remove the comment. Thanks.